### ISSUE TITLE
Discuss admin of creators and profiles

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/network-pulse-api#319]
Let's discuss any follow up work needed for how admins can associate creators and profiles. 

I have a few comments below on @gideonthomas's [PR notes](https://github.com/mozilla/network-pulse-api/pull/318#issue-172372422). In sum, 
  - I think the PR does the job of unblocking us on fellow profiles now. So that's good for now. I think we've  created new risks of user error. 
  - I wonder whether we need to step back and rethink the root of this. For example, if all existing and new Creators had a profile, empty and inactive, this might all be less complex. An active/inactive profile state seems easier to manage than shifting associations between models. 

I think we should push the PR after testing, and address the unintended consequences as a follow up with more time for all of us to think through things.  

Let's meet to discuss over video. A few rough notes for conversation (no need to reply): 
 - New Profiles can now be associated with existing Creators. This is great, and unblocks us. 
 - New and existing Profiles can be associated with Creators that already have a Profile. This is prone to user error and will create orphaned data an admin cannot correct. 
    - Admins should edit existing profiles rather than creating a new one. We risk duplicate profiles and unintentionally orphaned profiles still visible on the frontend. 
    - Admins could orphan a Creator. These creators will remain associated with entries and will be offered in  autocomplete on the new entry form.  

Lastly, I can't think of a use case for this:

> Update an existing profile and make sure that you can change its creator to the option that automatically creates a new creator.

**Comment by gideonthomas** [comment]
After discussing with @xmatthewx, here's a tentative plan for simplifying/optimizing creators:

### Changes
1. Remove the `Creator` and `OrderedCreator` models.
2. Whenever an entry has a creator, it is a profile. There will be no such thing as a "string" creator
3. Create an intermediary model between `UserProfile` and `Entry` to describe the many-to-many relationship between an entry and a creator (profile). This model will also have an order field to capture ordering of the creators.
4. When the front-end adds a creator to an entry that isn't a profile, on the back-end, we always create a profile (without a user) for that creator (effectively phasing out "string" creators) and mark the field `is_active` on that profile as `False`.
5. To populate auto-complete options on the front-end for creators, we send all profiles, active or inactive.
6. This will also give us the ability to re-associate "creators" with entries easily via the admin.
7. If an actual user creates a profile for themselves, we would have two different instances of a profile - one which is a creator on multiple entries but has no user attached to it and a second which has a user attached to it and is the "real" profile. We can then manually merge these profiles via the admin by re-associating the user with the "real" profile (and any entries associated with this profile, if any) with the profile that is a creator and then deleting the other profile. We might have a "merge profile" feature at some point that will automate this.

### Potential Action Plan
Start by removing all the admin code added to re-associate creators to profiles, then:
1. Create the intermediary model between `Entry` and `UserProfile`
2. Write a data migration to create profiles for every string creator
3. Write a data migration to represent data from the `Creator` and `OrderedCreatorRecord` models in the new intermediary model
4. Add logic to the entry serializer's `save` method to automatically create an inactive profile for creators that don't have a profile during deserialization
5. Remove the `Creator` and `OrderedCreatorRecord` models
6. Modify the admin form to reflect the changes to the model

@xmatthewx, have I missed/misrepresented anything from our discussion?

@cadecairos and @Pomax, what are your thoughts on this plan? Any concerns or comments?

**Comment by xmatthewx** [comment]
Thanks for the extensive notes @gideonthomas. In a nutshell, we remove complexity with two types of creators, and replace it with a simple active/inactive flag on a profile.

> Create an intermediary model between UserProfile and Entry to describe the many-to-many relationship

Do we continue to take a performance hit with this intermediary? 

> have I missed/misrepresented anything

We didn't discuss it, but someday in the future, we might want to give a new user the chance to say "hey, i'm the person listed on all those projects". Probably moderated. Definitely out of scope here.

**Comment by gideonthomas** [comment]
The intermediary model just allows us to fine tune the relationship between `Entry` and `UserProfile`. If we didn't use it, django automatically creates one for us. So the performance wouldn't be too different from us not creating this model.

**Comment by cadecairos** [comment]
This sounds sane to me.

**Comment by xmatthewx** [comment]
As we spec this out, let's think about admin features for manually or automatically merging profiles and re-associating entries.
